modules/ROOT/pages/dataweave-extension-plugin.adoc

```diff
@@ -0,0 +1,362 @@
+│       │           └── inputs //Folder where all of your inputs live
+│       │           │   └── payload.json //Input for your mapping (you can use it in your script, naming it to be the same as the name of the file without the file's extension)
+│       │           │   └── vars //Folder that represents a name for your input variable in your script: in this case ,vars
+│       │           │       └── test.json //Input for your mapping (you can use it in your script, calling it vars.test)
+│       │           └── out.json // Expected output of your mapping running against your inputs (only present when testing)
+```
+=== Using the DataWeave Scenarios View
+
+The *DATAWEAVE SCENARIOS* view shows the available scenarios for the `.dwl` file. In this view, you can add, remove, or set the active scenario that is used for your preview execution, or you can choose that the values be autocompleted.
+
+image::dataweave-extension-dataweavescenario.png[DataWeave Scenarios view in Visual Studio Code]
+
+
+== Preview a DataWeave Mapping
+
+After you define sample data, get feedback on how your mapping is working by previewing the output. To do so, follow one of these options:
+
+* Click the *DataWeave: Run Preview* button (arrow icon) at the top right corner of your mapping `.dwl` file window.
++
+image::dataweave-extension-preview-1.png[DataWeave: Run Preview button]
+
+* Right-click the editor and select *DataWeave: Enable AutoPreview* to run it on every file change.
++
+image::dataweave-extension-preview-2.png[DataWeave: Enable AutoPreview option]
+
+
+
+== Run and Debug a DataWeave Mapping
+
+Run and debug a DataWeave mapping to be guided through all of the mapping elements, which you can then fix or otherwise update. You can also set breakpoints to stop evaluating expressions and use all the debugging capabilities that Visual Studio Code provides. Debugging works on mapping tests as well.
+
+To run and debug a DataWeave mapping, follow these steps:
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/main/dw` folder.
+. Select your mapping `.dwl` file.
+. Click the *Run Mapping* button located at the top of your `.dwl` file.
++
+image::dataweave-extension-runmapping.png[MyMapping.dwl and Run Mapping button]
+
+
+== Test a DataWeave Mapping
+
+To test a DataWeave mapping, click the *Dataweave: Create Mapping Test* button located at the top of your `.dwl` file to create a new test scenario from a snapshot of your input and your current output.
+You can also use this newly created scenario as a normal scenario for your preview or autocompletion.
+
+To test a DataWeave mapping, follow these steps:
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/main/dw` folder.
+. Click the *DataWeave: Create Mapping Test* button (microscope icon) located at the top of your mapping `.dwl` file.
++
+image::dataweave-extension-testmapping.png[MyMapping.dwl file and the DataWeave Create Mapping Test button]
```

Comment:
Could we avoid the input directive in the image?
It would be nice to show the output of the preview here, so it is clear that we are saving the mapping scenario with the output of the Preview as a test.